velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -327,6 +329,84 @@ TpchPlan TpchQueryBuilder::getQ3Plan() const {
   return context;
 }
 
+// select
+// 	o.orderpriority,
+// 	count(*) as order_count
+// from
+// 	orders as o
+// where
+// 	o.orderdate >= '1993-07-01'
+// 	and cast(o.orderdate as date) < date '1993-07-01' + interval '3' month
+// 	and exists (
+// 		select
+// 			*
+// 		from
+// 			lineitem as l
+// 		where
+// 			l.orderkey = o.orderkey
+// 			and l.commitdate < l.receiptdate
+// 	)
+// group by
+// 	o.orderpriority
+// order by
+// 	o.orderpriority;
+
+TpchPlan TpchQueryBuilder::getQ4Plan() const {
+    std::vector<std::string> ordersColumns = {"o_orderdate", "o_orderpriority", "o_orderkey"};
+    std::vector<std::string> lineitemColumns = {"l_orderkey", "l_commitdate", "l_receiptdate"};
+    
+    auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+    const auto& ordersFileColumns = getFileColumnNames(kOrders);
+    auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+    const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+    
+    std::string orderDateFilter = formatDateFilter("o_orderdate", ordersSelectedRowType, "'1993-07-01'", "'1993-09-30'");
+    
+    auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+    core::PlanNodeId lineitemPlanNodeId;
+    core::PlanNodeId ordersPlanNodeId;
+
+    auto orders = PlanBuilder(planNodeIdGenerator, pool_.get())
+                        .tableScan(
+                            kOrders,
+                            ordersSelectedRowType,
+                            ordersFileColumns,
+                            {orderDateFilter})
+                        .capturePlanNodeId(ordersPlanNodeId)
+                        .planNode();
+
+    auto plan =
+        PlanBuilder(planNodeIdGenerator, pool_.get())
+            .tableScan(
+                kLineitem,
+                lineitemSelectedRowType,
+                lineitemFileColumns,
+                {},
+                "l_commitdate < l_receiptdate")
+            .capturePlanNodeId(lineitemPlanNodeId)
+            .finalAggregation({"l_orderkey"}, {}, {})
+            // .partialAggregation({"l_orderkey"}, {}, {})
+            // .finalAggregation()
+            .hashJoin(
+                {"l_orderkey"},
+                {"o_orderkey"},
+                orders,
+                "",
+                {"o_orderpriority", "o_orderkey"},
```

Comment:
Do we need `o_orderkey` column in the output?

Reply:
Will try without and see if it works and update if not will leave it as it is